### PR TITLE
Improve model

### PR DESCRIPTION
**Pull request by pmuir**
Many changes that start to remove hacks from the app

### CODE REVIEW

**Comment by pmuir** [comment]
Sorry, this is a large, somewhat tangled PR...

File: src/app/header/header.component.ts [hunk 1]
```diff
@@ -27,14 +27,46 @@ export class HeaderComponent implements OnInit {
...
+   };
+
+   public setActiveTab(index: number): void {
+   this.tabs[index].active = true;
+   };
+
+   public removeTabHandler(/!*tab:any*!/): void {
+   console.log('Remove Tab handler');
+   };
+   */
```
Comment (joshuawilson) [inline src/app/header/header.component.ts]: Did you intend to add this commented code back in or is it an artifact of the merge.

Comment (pmuir) [inline src/app/header/header.component.ts]: Merge artifact, good spot!

**Comment by pmuir** [comment]
@joshuawilson removed that added code

File: src/app/header/header.component.ts (hunk 1 above)
Comment (pmuir) [inline src/app/header/header.component.ts]: Fixed now

File: src/app/app.module.ts [hunk 2]
```diff
@@ -70,7 +72,11 @@ export type StoreType = {
     HttpModule,
     PublicModule,
     SpaceDialogModule,
-    AppRoutingModule
+    AppRoutingModule,
```
Comment (joshuawilson) [inline src/app/app.module.ts]: The AppRoutingModule needs to be the last thing in the list, everything else is usually listed alphabetically.

File: src/app/app.module.ts [hunk 3]
```diff
@@ -88,12 +94,13 @@ export type StoreType = {
     Logger,
     LoginService,
     ToggleService,
-    UserService
+    UserService,
+    ContextService
```
Comment (joshuawilson) [inline src/app/app.module.ts]: Please make this alphabetical

File: src/app/shared/context.service.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { Context } from './../models/context';
+import { DummyService } from './../dummy/dummy.service';
+import { Router } from '@angular/router';
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: This is a style thing but it is worth noting. We usually group the '@angular' imports first and then after a space have the local imports. If there are a lot of imports we tend to group them further based on the folder structure.

File: src/app/shared/context.service.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+/*
+ * A shared service that manages the users current context. The users context is defined as the
+ * entity (user or org) and space that they are operating on.
+ * 
+ */
+@Injectable()
+export class ContextService {
+
+    public current: Context;
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: Another style point. We use 2 space indents in most places, expect where a deeper indent improves readability.

File: src/app/signin/signin.component.ts [hunk 6]
```diff
@@ -13,15 +13,12 @@ export class SigninComponent implements OnInit {
...
-    
-  }
+  ngOnInit() {}
 
   signin() {
     // Fake a login
     localStorage.setItem('auth_token', 'pmuir');
-    // Another nice hack!
-    window.location.href = '/home';
+    this.router.navigate(['/home']);
```
Comment (joshuawilson) [inline src/app/signin/signin.component.ts]: Nice fix, thanks.

**Comment by pmuir** [comment]
New PR #40

**Pull request closed without merging**